x-pack/plugins/secops/public/components/page/hosts/hosts_table/index.tsx

```diff
@@ -132,7 +133,7 @@ const getHostsColumns = () => [
                   />
                 </DragEffects>
               ) : (
-                hostName
+                <Link to={`hosts/${host.hostId}`}>{hostName}</Link> // TODO: Does this need escapeQueryValue() or encodeURIComponent()?
```

Comment:
Few things here:
1. As the comment mentions - does this need wrapped in an `escapeQueryValue()` or `encodeURIComponent()`? What's the expected format of `hostId`? The ones in testing appeared to just be alphanumeric, but I noticed Xavier added `escapeQueryValue()` around the hostId he was using in the DraggableWrapper up above on line 121.
2. Should I be using the `EuiLink` component here?  If so, do you happen to know if [this method](https://github.com/elastic/eui/blob/master/wiki/react-router.md#1-conversion-function-recommended) outlined in the Eui/React Router docs is up to date? I only saw a few usages of that pattern when searching throughout the repo (few refs in `rollup`, `remote_clusters` and `cross_cluster_replication`)
3. Do I need a conditional here that doesn't create a link in case there is no `hostId`, or will `hostId` always be available?

Reply:
Regarding 1:

> 1. As the comment mentions - does this need wrapped in an escapeQueryValue() or encodeURIComponent()? What's the expected format of hostId? The ones in testing appeared to just be alphanumeric, but I noticed Xavier added escapeQueryValue() around the hostId he was using in the DraggableWrapper up above on line 121.

I don't know the expected format of hostId, but I think it makes sense to wrap and unwrap via `encodeURIComponent` / `decodeURIComponent`.

Reply:
Was just randomly reading through this. For all supported platforms (Linux, MacOS, Windows), the hostId is a UUID, though each platform has its own format:

- Linux is all lowercase, no hyphens: `6f7be6fb33e6c67f057266415c084408`
- Windows is all lowercase, with hyphens: `fa4a0df6-5a59-4870-84a0-cde535805dfc`
- MacOS is all uppercase, with hyphens: `155320EF-9DB6-5949-966B-F074A556AD32`

That being said, I don't think you can ever fully rely on a value in Elasticsearch, esp. unescaped. Nothing is stopping any client (Beats or otherwise) to write anything into it they want. It could be garbage, and it could be a malicious, e.g. JS code.

Reply:
It might make sense to normalize the hostId in the UI, e.g. convert all different UUID in the `host.id` field to one.

Reply:
I think that I will use EuiLink with the component linkTo do the job here, I mean the href should be something like that `#/link-to/hosts/678976ghjt65767`

Reply:
Updated the following:
- Wrapped `hostId` in encodeURIComponent
- Added conditional on `hostId` being present for link creation, and updated to use `EuiLink` with `#/link-to'
- As for normalization of `host.id` on the UI side, created https://github.com/elastic/ingest-dev/issues/203 and added an item to the SIEM Synch agenda to clarify

Reply:
Updated the following: 
- Wrapped `hostId` in `encodeURIComponent()` when creating the link to host details
- Added conditional to only create link if `hostId` isn't missing, and updated link to use `EuiLink` and `#/link-to`
- In reference normalizing the `hostId` on the UI side, I've created this ticket for discussion: https://github.com/elastic/ingest-dev/issues/203